lib/api/kuzzle.js

```diff
@@ -318,4 +317,24 @@ function dumpAndExit(kuzzle, suffix) {
   kuzzle.janitor.dump(suffix).finally(() => runShutdown(kuzzle));
 }
 
+function initVault(vaultKey, secretsFile) {
+  const defaultEncryptedSecretsFile = path.resolve(`${__dirname}/../../config/secrets.enc.json`);
+  const encryptedSecretsFile = secretsFile || process.env.KUZZLE_SECRETS_FILE || defaultEncryptedSecretsFile;
+
+  if (
+    (!_.isString(vaultKey) || vaultKey.length <= 0) &&
+    _.isString(process.env.KUZZLE_VAULT_KEY) &&
+    process.env.KUZZLE_VAULT_KEY.length > 0
+  ) {
+    vaultKey = process.env.KUZZLE_VAULT_KEY;
```

Comment:
Is this important?

Reply:
It's a minor issue, but it's considered bad practice to modify directly a function parameter directly because most of the time, without reading the whole function code, future maintainers are led to believe that the original function parameter is used, which is not the case.
This is a bit confusing (not much, that's why it's minor), so it should be avoided.

And since it costs almost nothing to fix, I think that it's a better way to tackle that issue rather than add a `// NOSONAR` comment.